org-code-javabuilder/lib/src/main/java/org/code/javabuilder/AWSSystemExitHelper.java

```diff
@@ -19,6 +19,8 @@ public AWSSystemExitHelper(String connectionId, AmazonApiGatewayManagementApi ap
 
   @Override
   public void exit(int status) {
+    PerformanceTracker.getInstance().trackInstanceEnd();
+    PerformanceTracker.getInstance().logPerformance();
```

Comment:
Alright, adding the performance tracker here ended up ballooning the PR a bit. Explanation for the additional changes:

In order to verify that the perf tracking was set up correctly for a system.exit situation, I tested out with our most recent occasion where we added a new System.exit. Namely, when a student recursively opens a file. However, I was blocked on testing this because we were sending the generic "We hit an error on our side" message to the user in this case. 

I spent some time investigating whether this was caused by my change and realized that instead, it was caused because we were unable to use the error handling classes because the class loader couldn't load them in the tmp directory due to the open file handles in the student's project. I was able to patch this by pre-loading those classes (See MainRunner.run).

However, I continued to hit an error where we were logging `SEVERE` errors to CloudWatch. Given that these cause alerts to show up in CloudWatch and Slack, I took a deeper look into these to see how these were being caused. I realized that since these are triggered by a rather innocuous-looking `logException` method, that method was being used in locations that shouldn't be triggering alerts. I was able to patch this by splitting `logException` into `logSevereException` and `logTrackingException`. (see LoggerUtils.logSevereException).

Reply:
@sanchitmalhotra126 @molly-moen - I could use a second glance since this added quite a bit to the PR. (see note)